plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -87,6 +87,38 @@ Blockly.Extensions.registerMutator('variable_entity_input',
         }
     });
 
+// Helper function to use in Blockly extensions that register one data list selector field to update contents of another
+// The block may define input called "<targetName>Field" to customize field's position
+// Note that the source field must be inserted before the target field for their values to be loaded properly
+function appendAutoReloadingDataListField(sourceName, targetName, targetList) {
+    return function () {
+        const thisBlock = this;
+        (this.getInput(targetName + 'Field') || this.appendDummyInput()).appendField(
+            new FieldDataListSelector(targetList, undefined, {
+                'customEntryProviders': function () {
+                    return thisBlock.getFieldValue(sourceName);
+                }
+            }), targetName);
+        this.setOnChange(function (changeEvent) {
+            if (changeEvent.type === Blockly.Events.BLOCK_CHANGE &&
+                changeEvent.blockId && changeEvent.getEventWorkspace_().getBlockById(changeEvent.blockId) === this &&

```

Comment:
Does comparing diretly by changeEvent.blockId and block.id not work?

If it does, you can avoid lookup of getBlockById as more lightweight solution

Reply:
Forgot there also is `block.id`

Reply:
> Maybe block id parameter could help uniquely identify the block to not update all of them (https://developers.google.com/blockly/reference/js/blockly.block_class.id_property.md)

Reply:
Like what I did?

Reply:
If it works (I did not test yet), yep :)